### DISCUSSION TITLE
Default schema v2 (and `linked_schemas` field improvement)

### DISCUSSION THREAD

**Discussion by geoffturk** [MurmurationsNetwork/MurmurationsProtocol#38] [Library (fields and schemas)]
The default schema is what is initially used by the Index to confirm a profile has at least one schema against which the profile will be validated.

Version 2 of the default schema was introduced in this commit:  
https://github.com/MurmurationsNetwork/MurmurationsLibrary/blob/2c4216b7aecb26c791623f61a6f19c7dacd9b692/schemas/default-v2.0.0.json

The regex has been improved to ensure lowercase text and to avoid any property names starting with a number:  
https://regex101.com/r/1ecQoV/3

**Comment by geoffturk** [comment]
New modifications for the property names regex:  
https://regex101.com/r/QmajQ9/4

Changes:
- Ensure property names do not start or end with an underscore (should only be letters or numbers)
- Adjust length values to ensure max property name length of 100 characters (0 to 97 = 98 characters, plus two more)

**Comment by geoffturk** [comment]
`linked_schemas` v0.0.1 => `^[a-z_][a-z0-9-._]{4,99}$`

`linked_schemas` v0.0.2 => `^[a-z][a-z0-9_]{7,97}-v[0-9]\.[0-9]\.[0-9]$`

https://regex101.com/r/bw1eF2/2

This tightens up the `linked_schemas` naming requirements to stick to the following format:  
`some_schema-v1.0.0`

Implementing this change means that the old `-v1` schemas will no longer be usable. We should aim to update those to `-v2.0.0` by adding the newer field definitions/

**Comment by Photosynthesis** [comment]
Semver major, minor and patch version numbers can each be multiple characters (eg v11.9.10), so I think there should be a `+` after the numeric range patterns in the regex (eg `^[a-z][a-z0-9_]{7,97}-v[0-9]+\.[0-9]+\.[0-9]+$`).

**Comment by geoffturk** [comment]
Good point. Updated:  
https://regex101.com/r/bw1eF2/3

**Comment by geoffturk** [comment]
We moved validation logic of indexed fields out of the code base and into JSON Schema in the default schema, as documented in this issue: https://github.com/MurmurationsNetwork/MurmurationsServices/issues/827

**Comment by geoffturk** [comment]
And we added a new field to the default schema that controls the automatic deletion of a profile when it reaches a specific date/time: https://github.com/MurmurationsNetwork/MurmurationsServices/issues/829
